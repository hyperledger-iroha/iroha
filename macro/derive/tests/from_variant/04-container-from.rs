include!("03-container-enums.rs");

impl std::convert::From<Variant1> for Enum {
    fn from(variant: Variant1) -> Self {
        Self::Variant1(Box::new(variant))
    }
}
