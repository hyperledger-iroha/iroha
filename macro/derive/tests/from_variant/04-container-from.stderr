error[E0119]: conflicting implementations of trait `std::convert::From<Variant1>` for type `Enum`
  --> $DIR/03-container-enums.rs:16:10
   |
16 | #[derive(iroha_derive::FromVariant)]
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Enum`
   |
  ::: $WORKSPACE/macro/derive/tests/from_variant/04-container-from.rs
   |
   | impl std::convert::From<Variant1> for Enum {
   | ------------------------------------------ first implementation here
   |
   = note: this error originates in the derive macro `iroha_derive::FromVariant` (in Nightly builds, run with -Z macro-backtrace for more info)
